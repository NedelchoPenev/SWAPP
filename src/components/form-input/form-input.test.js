import React from 'react';
import { shallow } from 'enzyme';

import FormInput from './form-input.component';

describe('FormInput component', () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      value: 'test@gmail.com',
      handleChange: mockHandleChange
    };

    wrapper = shallow(<FormInput {...mockProps} />);
  });

  it('should render FormInput component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange method when input changes', () => {
    wrapper.simulate('change');

    expect(mockHandleChange).toHaveBeenCalled();
  });
});