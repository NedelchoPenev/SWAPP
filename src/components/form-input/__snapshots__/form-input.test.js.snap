// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render FormInput component 1`] = `
<styled.input
  onChange={[MockFunction]}
  value="test@gmail.com"
/>
`;
